db/schema_tables.cc

```diff
@@ -947,11 +950,34 @@ static future<> with_merge_lock(noncopyable_function<future<> ()> func) {
     }
 }
 
+template <typename Func>
+requires std::same_as<futurize_t<std::invoke_result_t<Func, replica::database&>>, future<>>
+static future<> update_database(sharded<replica::database>& db, Func func) {
+    using futurator = futurize<std::invoke_result_t<Func, replica::database&>>;
+    co_await futurator::invoke(func, db.local());
+    try {
+        co_await db.invoke_on_others(func);
```

Comment:
It could be any shard for that matter, for the purpose of "seeding" a new effective replication map.
Other shards should find the new effective replication map, if one was created, and use it.

Reply:
Ah, so this is just an optimization?

Please document that.

Reply:
But those listeners are not only used for erm replication. The promise made by the notifier is that the listener is called on each shard. Some notification handlers can rely on the state on the local shard and not take it from shard 0. Some of them invalidate shard-local caches, for example.

If the state is taken from shard 0, it's better not to have each shard fetch the state from shard 0 (seed), but handle the notification only on shard 0 and have the handler replicate to other shards explicitly (like in replicate_to_all_cores()).

Reply:
the listeners are still notified on all shards, just in a 2nd pass.